source/development/feature-flags.html.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Feature flags in GOV.UK Frontend
+weight: 1
+last_reviewed_on: 2023-11-28
+review_in: 6 months
+---
+# Using feature flags in GOV.UK Frontend
+
+Feature flags are a way for us to release large scale or experimental work early by allowing users to opt in our out of a new, breaking change or feature. Use this guidance to decide if a feature flag is useful for your use case.
+
+## When a feature flag is useful
+Feature flags are useful when the following conditions are true.
+
+### The thing is a breaking change
+Feature flags should only be used for things considered breaking changes. This is something where users will potentially need to change their service, or assess if their service needs changing, in order to continue staying up to date with govuk-frontend. In this case, a feature flag would be a way for a user to stay up to date whilst also opting out of the feature for a period of time. We don’t need to do this for minor releases or new features such as a new component because a user can opt out by simply not using the new feature.
+
+### The thing might be a long term project for users
+When assessing if a breaking change should be behind a feature flag, consider the scope of work required by the user. A feature flag is useful if the change is very difficult to implement and/or has wide scope. Using the green link example in the previous section, this would impact all links in a given service so that service will likely need to check all their pages to ensure the design of their pages still made sense against the new link colour.
+
+Feature flags are more useful for wide scope changes that are likely to take a long time to action over difficult to implement changes. For difficult to implement but short term changes, consider using a pre-release instead (see below section on pre-releases).
+
+### We are unsure of the thing’s value
+This is an alternative to the previous principal on implementation difficulty where, for whatever reason, we have a significant change that we aren’t 100% confident in and have been unable to verify its value. In this instance, putting it behind a feature flag would allow us to get early feedback from users who’re willing to use and test our new/updated feature and give us feedback, allowing us to iterate the change with data.
+
+## When a feature flag isn't useful
+Even if you have a scenario where feature flags may be useful, consider the following before implementing one.
+
+### When releasing the thing carries no significant risk
+We should avoid using feature flags in most cases. We have alternative systems for communicating big or impactful changes such as feature depreciation where we tell people we’ll be removing something in a future breaking release and breaking releases themselves where users know that by upgrading, they might need to change something in their service. Feature flags should be used as a way to alleviate instances where releasing as normal is challenging.
```

Comment:
Q: do you want all mentions of the feature flag to be first word capitalised, both words capitalised or all lowercase unless starting a sentence?

Reply:
I personally think all lowercase. Does that feel sensible?

Reply:
let's try it. we can always change it later if it doesn't work for some reason.